### ISSUE TITLE
Migrate some script to modern JavaScript

### ISSUE THREAD

**Issue by FranklinYu** [MacDownApp/macdown#959]
@uranusjr has mentioned in <https://github.com/MacDownApp/macdown/pull/958#discussion_r180462883> that we may convert/rewrite some script with modern JavaScript (i.e. ES6+). We need transpiler (such as Bable) because according to [this post](https://stackoverflow.com/questions/42374413/how-do-i-set-the-webkit-version-used-by-a-webview), WebView in macOS 10.11 uses WebKit 601.7.8 (i.e. Safari 9, according to [Wikipedia](https://en.wikipedia.org/wiki/Safari_version_history)), which [supports virtually nothing in ES6](https://kangax.github.io/compat-table/es6/#safari9).

Simple Babel config should be OK:

```json
{
  "presets": ["env"]
}
```

This would transform latest JavaScript to ES5, which (I assume) works well even in WebKit 536.30.1 in macOS 10.8.

Still I'm not sure whether it's necessary to migrate all JavaScript files. In addition, #952 may need modification or rework, such as renaming the folder to `asset-generator`.

**Comment by franklinyu** [comment]
@uranusjr Yes, but the WebKit for WebView will not upgrade even if user upgrade their Safari, as explained in [this answer](https://stackoverflow.com/questions/42374413/how-do-i-set-the-webkit-version-used-by-a-webview/42658333#42658333). Safari will bundle its own WebKit to run on older platforms.

> New major Safari versions not only add shiny new features, and bugs, they sometimes also remove things. Some ill faded `-webkit` prefixes come to mind. WebKit is part of the SDK which *guaranties* you that your app will not break on such updates.

**Comment by uranusjr** [comment]
OMG
